populus/cli/attach_cmd.py

```diff
@@ -12,29 +14,172 @@
     is_ipython = False
 
 from eth_rpc_client import Client
+from watchdog.events import FileSystemEventHandler
 
 import populus
 from populus import utils
 from populus.contracts import (
     package_contracts,
 )
-
+from populus.geth import (
+    get_active_data_dir,
+    get_known_contracts,
+    add_to_known_contracts,
+)
+from populus.deployment import (
+    deploy_contracts,
+    validate_deployed_contracts,
+)
+from populus.cli.deploy_cmd import (
+    echo_post_deploy_message,
+)
+from populus.observers import (
+    get_active_dir_observer,
+)
 from .main import main
 
 
+def deploy_set(context, client, project_dir, data_dir=None, record=True, contracts_by_name=[]):
+    """ Re-Deploy a set of contracts by name or all contracts
+        @param context local context for the attach python interpreter.
+        @param client RPC client for accessing the ethereum chain
+        @param project_dir absolute file path of the project that
+            will be interrogated to determine the contracts that
+            can be deployed
+        @param data_dir directory of the test chain if not None
+        @param contracts_by_name if not None, this must be a list
+            of strings indicating the names of the contracts to deploy
+    """
+
+    if type(contracts_by_name) is not list:
+        raise TypeError("Contracts by name must be a list of strings")
+
+    contracts = package_contracts(utils.load_contracts(project_dir))
+    deployed_contracts = deploy_contracts(
+        deploy_client=client,
+        contracts=contracts,
+        deploy_at_block=1,
+        max_wait_for_deploy=120,
+        from_address=None,
+        max_wait=120,
+        contracts_to_deploy=contracts_by_name,
+        dependencies=None,
+        constructor_args=None,
+        deploy_gas=None,
+    )
+    validate_deployed_contracts(client, deployed_contracts)
+    echo_post_deploy_message(client, deployed_contracts)
+    if data_dir is not None and record:
+        add_to_known_contracts(deployed_contracts, data_dir)
+    # Update the attach shell's context with the latest contracts
+    # objects
+    context["contracts"] = contracts
+    if data_dir is not None:
+        setup_known_instances(context, data_dir)
+    return(deployed_contracts)
+
+
+def setup_known_instances(context, data_dir):
+    # Attempt to load known contracts.
+    knownCts = get_known_contracts(data_dir)
+    for name, ctType in context["contracts"]:
+        if name in knownCts.keys():
+            addrList = knownCts[name]
```

Comment:
I would prefer that the naming conventions for the local variables be in the format of `known_cts` to remain consistent with the rest of the codebase.